File: Graph/php/largestComponent.php
<?php

declare(strict_types=1);

function largestComponent(array $graph): int
{
    $visited = [];
    $largest = 0;

    foreach ($graph as $node => $_) {
        $size = componentSize($graph, (string) $node, $visited);
        if ($size > $largest) {
            $largest = $size;
        }
    }

    return $largest;
}

function componentSize(array $graph, string $node, array &$visited): int
{
    if (isset($visited[$node])) {
        return 0;
    }
    $visited[$node] = true;

    $size = 1;
    foreach ($graph[$node] ?? [] as $neighbor) {
        $size += componentSize($graph, (string) $neighbor, $visited);
    }
    return $size;
}

File: Graph/go/largest_component.go
package graph

// LargestComponent returns size of the largest connected component.
func LargestComponent(g Graph) int {
	visited := map[string]struct{}{}
	maxSize := 0
	for node := range g {
		size := componentSize(g, node, visited)
		if size > maxSize {
			maxSize = size
		}
	}
	return maxSize
}

func componentSize(g Graph, node string, visited map[string]struct{}) int {
	if _, seen := visited[node]; seen {
		return 0
	}
	visited[node] = struct{}{}
	size := 1
	for _, neighbor := range g[node] {
		size += componentSize(g, neighbor, visited)
	}
	return size
}
